openjdk/mmtkBarrierSet.hpp

```diff
@@ -38,6 +38,8 @@
 
 #define MMTK_ENABLE_ALLOCATION_FASTPATH true
 
+const intptr_t ALLOC_BIT_BASE_ADDRESS = GLOBAL_SIDE_METADATA_BASE_ADDRESS;
```

Comment:
You are assuming the first global side metadata spec is always the global alloc bit. It is true for now, but there is no way to guarantee that this won't be changed. Like in your alloc bit PR, you put the alloc bit metadata to the first. Others may have this kind of change as well. You need the address for the alloc bit.